wrapper/Convert/SireOpenMM/sire_to_openmm_system.cpp

```diff
@@ -398,6 +400,130 @@ void _add_positional_restraints(const SireMM::PositionalRestraints &restraints,
     }
 }
 
+/** Add all of the angle restraints from 'restraints' to the passed
+ *  system, which is acted on by the passed LambdaLever. The number
+ *  of real (non-anchor) atoms in the OpenMM::System is 'natoms'
+ */
+
+void _add_angle_restraints(const SireMM::AngleRestraints &restraints,
+                           OpenMM::System &system, LambdaLever &lambda_lever,
+                           int natoms)
+{
+    if (restraints.isEmpty())
+        return;
+
+    // energy expression of the angle restraint, which acts over three atoms
+    //
+    // theta = angle(P1, P2, P3)
+    //
+    // The energies are
+    //
+    // e_restraint = rho * (e_angle)
+    // e_angle = ktheta(theta - theta0)^2
+
+    const auto energy_expression = QString(
+                                       "rho*k*(theta-theta0)^2;")
+                                       .toStdString();
+
+    auto *restraintff = new OpenMM::CustomAngleForce(energy_expression);
+
```

Comment:
All of the custom forces (the ones we add on top of the vanilla OpenMM ones) are now named. This makes it easy to search for them, or identify them in an XML file. This is particularly true when different forces are of the same type, e.g. `CustomBondForce`, but do different things. Please add names for your new restraints using `restraintff->setName()`. Take a look at the other `_add_*_restraints` functions to see how this is done. Something like the following:

```c++
    auto *restraintff = new OpenMM::CustomAngleForce(energy_expression);
    restraintff->setName("AngleRestraintForce")
```

Reply:
Thanks @akalpokas, I'll try to take another look through by the end of the week.